src/misc/host_locale_win32.cpp

```diff
@@ -358,10 +358,12 @@ static std::string to_string(const wchar_t* input, const size_t input_length)
 		return {};
 	}
 
-	std::vector<char> buffer(buffer_size);
+	std::vector<char> buffer(buffer_size + 1);
 	WideCharToMultiByte(DefaultCodePage,  0, input, check_cast<int>(input_length),
 	                    buffer.data(), buffer_size, nullptr, nullptr);
-	return std::string(buffer.begin(), buffer.end());
+	buffer[buffer_size] = '\0';
+
+	return std::string(buffer.begin(), std::find(buffer.begin(), buffer.end(), '\0'));
 }
```

Comment:
Does not work, unfortunately.

Reason: both `num_converted` and `buffer_size` are.. 85 (sic!), at least on my system. We need to trim the result somehow, otherwise the `std::map<std::string, ...>::contains(const std::string &key)` function gets confused and fails to find the data entry. Even though `LOG("Host value is '%s'", result.c_str());` does not show anything suspicious, as the string is like `"pl-PL\0(whatever garbage Windows decided to put here)"` and C code considers the `\0` as the end of the string.